compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -1243,6 +1252,21 @@ impl<'a: 'ast, 'ast> LateResolutionVisitor<'a, '_, 'ast> {
         }
     }
 
+    fn dectect_wrong_primitive(path: &[Segment]) -> Option<Symbol> {
+        let name = path[path.len() - 1].ident.name;
+        // common java types
+        match name {
```

Comment:
This is up to you, but I prefer to wrap the entire `match` in `Some`, and then `return None` in the `_` branch, which leads to less repetition of `Some`.